src/lib/LoginFile.js

```diff
@@ -0,0 +1,274 @@
+/* global QrEncoder */
+
+class LoginFile {
+    /**
+     * @param {string} encodedPrivKey
+     * @param {number} [color]
+     */
+    constructor(encodedPrivKey, color = 0) {
+        this._width = LoginFile.WIDTH;
+        this._height = LoginFile.HEIGHT;
+        const $canvas = document.createElement('canvas');
+        $canvas.width = this._width;
+        $canvas.height = this._height;
+        this.$canvas = $canvas;
+        this._config = LoginFile.CONFIG[color];
+        if (!this._config) throw new Error(`Invalid color index: ${color}`);
+        /** @type {CanvasRenderingContext2D} */
+        this._ctx = ($canvas.getContext('2d'));
+        this._drawPromise = this._draw(encodedPrivKey);
+    }
+
+    static calculateQrPosition() {
+        return {
+            x: 69,
+            y: 287,
+            size: LoginFile.QR_SIZE,
+            padding: LoginFile.QR_PADDING,
+            width: LoginFile.QR_BOX_SIZE,
+            height: LoginFile.QR_BOX_SIZE,
+        };
+    }
+
+    filename() {
+        const walletName = 'Nimiq Login File '
+                         + `${this._config.name[0].toUpperCase()}${this._config.name.substr(1)} Wallet`;
+        return `${walletName.replace(/ /g, '-')}.png`;
+    }
+
+    async toDataUrl() {
+        await this._drawPromise;
+        return this.$canvas.toDataURL().replace(/#/g, '%23');
+    }
+
+    async toObjectUrl() {
+        await this._drawPromise;
+
+        return new Promise(resolve => {
+            this.$canvas.toBlob(blob => {
+                const url = URL.createObjectURL(blob);
+                resolve(url);
+            });
+        });
+    }
+
+    /**
+     * @param {string} encodedPrivKey
+     * @returns {Promise<void>}
+     */
+    async _draw(encodedPrivKey) {
```

Comment:
```suggestion
    async _draw(encodedPrivateKey) {
```
no need to shorten the name.